lib/timers.js

```diff
@@ -554,67 +557,73 @@ function tryOnImmediate(immediate, queue) {
   }
 }
 
+function runCallback(timer) {
+  var argv = timer._argv;
+  var argc = argv ? argv.length : 0;
+  switch (argc) {
+    // fast-path callbacks with 0-3 arguments
+    case 0:
+      return timer._callback();
+    case 1:
+      return timer._callback(argv[0]);
+    case 2:
+      return timer._callback(argv[0], argv[1]);
+    case 3:
+      return timer._callback(argv[0], argv[1], argv[2]);
+    // more then 3 arguments run slower with .apply
+    default:
+      return timer._callback.apply(timer, argv);
+  }
+}
 
-function Immediate() { }
-
-Immediate.prototype.domain = undefined;
-Immediate.prototype._onImmediate = undefined;
-Immediate.prototype._idleNext = undefined;
-Immediate.prototype._idlePrev = undefined;
 
+function createImmediate(callback, args) {
```

Comment:
it's weird, I just ran a set of micro-benchmarks and they're exactly like you say,
setting `this.x` properties is very fast, and 50% faster than inheriting from `prototype`.
But inside timers.js a { ... } style struct is faster.

I ran the same benchmark side-by-side with --trace-gc, and the `new Immediate` version
prints 20% more gc scavenge lines (207 vs 248).  But it also runs 20% longer, 5200
vs 6250 ms, so the gc may be normal (proportional to runtime), not sure.  The only change
was how the immediate objects are allocated, `createImmediate` vs `new Immediate`
with the object
```
function Immediate(callback, args) {
    this._idleNext = null;
    this._idlePrev = null;
    this._callback = callback;
    this._argv = args;
    this._onImmediate = callback;
    this.domain = process.domain;
}
```

The timers/immediate.js breadth benchmark shows a larger 55% difference in favor of { ... }
(the 0 and 1 arg cases; the 4-args case is 11%)